### PR TITLE
[Metricbeat] Migrate Kubernetes state_pod Metricset to use ReporterV2 interface

### PR DESCRIPTION
**Pull request by sayden**
Refer to elastic/beats#10774 for more info

### CODE REVIEW

**Comment by sayden** [comment]
jenkins, test this

**Comment by sayden** [comment]
Error in ES module seem unrelated https://beats-ci.elastic.co/job/elastic+beats+pull-request+multijob-linux/beat=metricbeat,label=linux-immutable/5455/console

**Comment by sayden** [comment]
After last push, error is related. Checking

**Comment by sayden** [comment]
I'm not sure if `state_pod_test.go` unit test can be removed already that we are using the new testing framework. @ruflin you are the one who knows better what's covered by it, could you confirm or not my guess?

File: metricbeat/module/kubernetes/state_pod/_meta/testdata/kube-state-metrics-v1.3.0.plain-expected.json [hunk 1]
```diff
@@ -0,0 +1,284 @@
+[
+    {
+        "event": {
+            "dataset": "kubernetes.pod",
+            "duration": 115000,
+            "module": "kubernetes"
+        },
+        "kubernetes": {
+            "pod": {
+                "_module": {
```
Comment (jsoriano) [inline metricbeat/module/kubernetes/state_pod/_meta/testdata/kube-state-metrics-v1.3.0.plain-expected.json]: What are these `_module` objects? I think they shouldn't be here.

File: metricbeat/module/kubernetes/state_pod/_meta/testdata/kube-state-metrics-v1.3.0.plain-expected.json [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+                    "namespace": "kube-system"
```
Comment (jsoriano) [inline metricbeat/module/kubernetes/state_pod/_meta/testdata/kube-state-metrics-v1.3.0.plain-expected.json]: Kubernetes namespace should be in `kubernetes.namespace`.

File: metricbeat/module/kubernetes/state_pod/_meta/testdata/kube-state-metrics-v1.3.0.plain [hunk 3]
```diff
@@ -0,0 +1,464 @@
+# HELP go_gc_duration_seconds A summary of the GC invocation durations.
```
Comment (jsoriano) [inline metricbeat/module/kubernetes/state_pod/_meta/testdata/kube-state-metrics-v1.3.0.plain]: Is this the same file as the one in `metricbeat/module/kubernetes/_meta/test/`?

Could we also add `metricbeat/module/kubernetes/_meta/test/kube-state-metrics` to this testdata? If we can I think we could remove `state_pod_test.go` as you suggested.

File: metricbeat/module/kubernetes/state_pod/_meta/testdata/kube-state-metrics-v1.3.0.plain-expected.json [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+            "dataset": "kubernetes.pod",
+            "duration": 115000,
+            "module": "kubernetes"
+        },
+        "kubernetes": {
+            "pod": {
+                "_module": {
+                    "namespace": "kube-system",
+                    "node": {
+                        "name": "minikube"
```
Comment (jsoriano) [inline metricbeat/module/kubernetes/state_pod/_meta/testdata/kube-state-metrics-v1.3.0.plain-expected.json]: Kubernetes node name should be in `kubernetes.node.name`.

File: metricbeat/module/kubernetes/state_pod/state_pod.go [hunk 5]
```diff
@@ -94,17 +89,28 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
...
+		m.Logger().Error(err)
+		reporter.Error(err)
+		return
 	}
-	return events, err
+
+	m.enricher.Enrich(events)
+	for _, event := range events {
+		reporter.Event(mb.Event{
+			MetricSetFields: event,
```
Comment (jsoriano) [inline metricbeat/module/kubernetes/state_pod/state_pod.go]: You should use `RootFields` for common fields as `kubernetes.namespace` or `kubernetes.pod.name`.

**Pull request merged**